src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -57,6 +57,51 @@ The .NET Foundation licenses this file to you under the MIT license.
       <StandaloneGCSupportName Condition="'$(IlcStandaloneGCSupport)' == 'true'">libstandalonegc-enabled</StandaloneGCSupportName>
     </PropertyGroup>
 
+    <PropertyGroup Condition="'$(_iOSLibraryMode)' == 'true'">
```

Comment:
Should this be in a stand alone task?

Reply:
> Should this be in a stand alone task? At least in some form?

Do you mean to group these in a separate MSBuild target? 
- We could, but then we would have to escape setting up the following as they would occur after the new target https://github.com/dotnet/runtime/blob/b0e7ee7c8d165e49b99113e23e08d2d1c17f88a2/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets#L43-L47

- Additionally, this new target (lets call it `SetupAppleSpecificProps`) should be executed before `SetupOSSpecificProps` and I already had some attempts to make that work properly see: discussion above https://github.com/dotnet/runtime/pull/93658#discussion_r1364994941 
Basically the problem is that we already have some external targets hooking in before `SetupOSSpecificProps`, so if it happens that these external targets are also trying to set properties before `SetupAppleSpecificProps` there is no guarantee that the external target will be scheduled by MSBuild before this newly introduced one.